PurchasesCoreSwift/Attribution/AttributionPoster.swift

```diff
@@ -0,0 +1,167 @@
+@objc(RCAttributionPoster) public class AttributionPoster: NSObject {
+
+    let deviceCache: DeviceCache
+    let identityManager: IdentityManager
+    let backend: Backend
+    let systemInfo: SystemInfo
+    let attributionFetcher: AttributionFetcher
+    let subscriberAttributesManager: SubscriberAttributesManager
+
+    private static var postponedAttributionData: [AttributionData]?
+
+    @objc public init(deviceCache: DeviceCache,
+                      identityManager: IdentityManager,
+                      backend: Backend,
+                      systemInfo: SystemInfo,
+                      attributionFetcher: AttributionFetcher,
+                      subscriberAttributesManager: SubscriberAttributesManager) {
+        self.deviceCache = deviceCache
+        self.identityManager = identityManager
+        self.backend = backend
+        self.systemInfo = systemInfo
+        self.attributionFetcher = attributionFetcher
+        self.subscriberAttributesManager = subscriberAttributesManager
+    }
+
+    @objc(postAttributionData:fromNetwork:forNetworkUserId:)
+    public func post(attributionData data: [String: Any],
+                     fromNetwork network: AttributionNetwork,
+                     forNetworkUserId networkUserId: String?) {
+        Logger.debug(Strings.attribution.instance_configured_posting_attribution)
+        if data["rc_appsflyer_id"] != nil {
+            Logger.warn(Strings.attribution.appsflyer_id_deprecated)
+        }
+
+        if network == .appsFlyer && networkUserId == nil {
+            Logger.warn(Strings.attribution.networkuserid_required_for_appsflyer)
+        }
+
+        guard let appUserID = self.identityManager.maybeCurrentAppUserID else {
+            Logger.error(Strings.attribution.missing_app_user_id)
+            return
+        }
+
+        let identifierForAdvertisers = attributionFetcher.identifierForAdvertisers
+        if identifierForAdvertisers == nil {
+            Logger.warn(Strings.attribution.missing_advertiser_identifiers)
+        }
+
+        let networkKey = String(network.rawValue)
+        let dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks = deviceCache.latestNetworkAndAdvertisingIdsSent(appUserID: appUserID)
+        let latestSentToNetwork = dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks[networkKey]
+
+        // TODO: `(null)` is true to the ObjC code here, maybe we should reject this and not post?
+        let newValueForNetwork = "\(identifierForAdvertisers ?? "(null)")_\(networkUserId ?? "(null)")"
+        guard latestSentToNetwork != newValueForNetwork else {
+            Logger.debug(Strings.attribution.skip_same_attributes)
+            return
+        }
+
+        var newDictToCache = dictOfLatestNetworkIdsAndAdvertisingIdsSentToNetworks
+        newDictToCache[networkKey] = newValueForNetwork
+        var newData = data
+
+        newData["rc_idfa"] = identifierForAdvertisers
+        newData["rc_idfv"] = attributionFetcher.identifierForVendor
```

Comment:
this _might_ actually not be a direct translation of objective-c: 
in objc, if you do `dict[@"foo"] = nil`, that'll _remove_ the key, whereas in swift, it'll add an optional value. 
So we probably wanna turn these into if-lets to do a 1-1 translation

Reply:
Great catch.